ClientApp/viking-vault/src/TransferFunds/TransferFundsPage.tsx

```diff
@@ -0,0 +1,328 @@
+    cardNumber: string;
+    transferDetails: string;
+    modalMessage: string;
+    totalBalance: number;
+    openModal: boolean;
+    requestTransfer: boolean;
+    isButtonDisabled: boolean;
+}
+
+class TransferFundsPage extends React.Component<any, ITransferFundsState>{
+    constructor(props:any){
+        super(props);
+        this.state = {
+          transferedAmount: 0,
+          currency: "EUR",
+          cardNumber: "",
+          transferDetails: "",
+          modalMessage: "",
+          totalBalance: 0,
+          openModal: false,
+          requestTransfer: false,
+          isButtonDisabled: false
+        }
+    }
+
+    componentDidMount(){
+      this.getTotalBalance();      
+    }
+
+    componentDidUpdate() {
+      this.getTotalBalance();
+    }
+
+    setAmountToBeTransfered = (e : any) =>
+    {
+        if(Number (e.target.value) <= this.state.totalBalance) {
+          this.setState({
+            transferedAmount: e.target.value
+          });
+        }
+    }
+
+    handleChangedCardNumber = (e : any) =>
+    {
+        this.setState({
+            cardNumber: e.target.value
+        })
+    }
+
+    handleChangedTransferDetails = (e : any) =>
+    {
+        this.setState({
+            transferDetails: e.target.value
+        })
+    }
+
+    private setCurrency = (e: any) => {
+      this.setState({
+        currency: e.target.value
+      });
+      this.getTotalBalance();
+    }
+
+    handleChangerequestTransferState = () =>
+    {
+      this.setState((oldState: ITransferFundsState) => ({
+        requestTransfer: !oldState.requestTransfer
+      }));
+        
+    }
+
+    handleIsButtonDisabled = () =>
+    {
+        this.setState((oldState: ITransferFundsState) => ({
+          isButtonDisabled: !oldState.isButtonDisabled
+        }));
+    }
+
+    getTotalBalance = () => {
+      let token = sessionStorage.getItem('Authentication-Token');
+        if(token != null) {
+            fetch(constants.baseUrl+"bankAccount", {
+                method: "GET",
+                headers: {
+                  'Accept': 'application/json',
+                  'Content-Type': 'application/json',
+                  'x-access-token': token.toString()
+                }
+            })
+            .then(response => response.json())
+            .then(result => {
+                if(this.state.currency === "RON") {
+                    this.setState({
+                        totalBalance: result[0].balance.toFixed(2)
+                    });
+                }
+                if(this.state.currency === "EUR") {
+                    this.setState({
+                        totalBalance: result[1].balance.toFixed(2)
+                    });
+                }
+                if(this.state.currency === "USD") {
+                    this.setState({
+                        totalBalance: result[2].balance.toFixed(2)
+                    });
+                }
+                if(this.state.currency === "YEN") {
```

Comment:
Please find a way to get rid of these multiple if's, they're really a bad practice as if they are hard to modify and it's also a lot of repeating code.

Proposed solution
-- create a dictionary for currencies and their indexes (or indices, whatever it's spelled) in the result array
-- return something like this `result[currenyMap[this.state.currency]].balance.toFixed(2)`
(of course also handle the default value, you can use Object.HasOwnProperty("propName") too see if an object has a property with that name)